libdnf/dnf-repo.cpp

```diff
@@ -1066,13 +1085,30 @@ dnf_repo_set_keyfile_data(DnfRepo *repo, GError **error)
     /* gpgkey is optional for gpgcheck=1, but required for repo_gpgcheck=1 */
     g_strfreev(priv->gpgkeys);
     priv->gpgkeys = NULL;
-    tmp_strval = g_key_file_get_string(priv->keyfile, repoId, "gpgkey", NULL);
-    if (tmp_strval) {
-        priv->gpgkeys = g_strsplit_set(tmp_strval, " ,", -1);
-        g_free(g_steal_pointer (&tmp_strval));
-        /* Canonicalize the empty list to NULL for ease of checking elsewhere */
-        if (priv->gpgkeys && !*priv->gpgkeys)
+
+    g_auto(GStrv) gpgkeys;
+    gpgkeys = g_key_file_get_string_list(priv->keyfile, repoId, "gpgkey", NULL, NULL);
+
+    if (gpgkeys) {
+        // gpgkeys can be ['value1', 'value2, value3'] therefore we first join to have 'value1, value2, value3'
+        tmp_strval = g_strjoinv(",", gpgkeys);
```

Comment:
What about don't use/remove "global" `tmp_strval`. Instead it can be defined at places where it is used.
More info in comment for baseurl.